clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -586,7 +587,7 @@ private KafkaConsumer(ConsumerConfig config,
                     config.getInt(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG), time);
             this.client = new ConsumerNetworkClient(netClient, metadata, time, retryBackoffMs,
                     config.getInt(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG));
-            OffsetResetStrategy offsetResetStrategy = OffsetResetStrategy.valueOf(config.getString(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG).toUpperCase());
+            OffsetResetStrategy offsetResetStrategy = OffsetResetStrategy.valueOf(config.getString(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG).toUpperCase(Locale.ROOT));
```

Comment:
@omkreddy  `Locale.setDefault(..)` changes the default locale for the whole JVM. This can have unintended consequences in the client application, for example with user-visible strings. Permissions for `Locale.setDefault(..)` may also be denied by the Java security manager depending on the Java security configuration, hence we cannot assume that the operation will succeed.